src/jv_print.c

```diff
@@ -229,16 +231,29 @@ static void jv_dump_term(struct dtoa_context* C, jv x, int flags, int indent, FI
     put_str("true", F, S, flags & JV_PRINT_ISATTY);
     break;
   case JV_KIND_NUMBER: {
-    double d = jv_number_value(x);
-    if (d != d) {
-      // JSON doesn't have NaN, so we'll render it as "null"
-      put_str("null", F, S, flags & JV_PRINT_ISATTY);
+    if (jvp_number_is_nan(x)) {
+      jv_dump_term(C, jv_null(), flags, indent, F, S);
     } else {
-      // Normalise infinities to something we can print in valid JSON
-      if (d > DBL_MAX) d = DBL_MAX;
-      if (d < -DBL_MAX) d = -DBL_MAX;
-      put_str(jvp_dtoa_fmt(C, buf, d), F, S, flags & JV_PRINT_ISATTY);
+#ifdef USE_DECNUM
+      const char * literal_data = jv_number_get_literal(x);
```

Comment:
Do you need to check that `jv_number_has_literal(x)` before calling `jv_number_get_literal(x)`?

Reply:
nope, it will return NULL if no literal is available on the number.

Reply:
Ok.  I had taken a brief look and gotten the impression it would assert that the number has a literal.  I'll figure that out for sure this weekend.

Reply:
It asserts that it is a _number_, but gracefully checks for literal value presence.